### PR TITLE
fix(developer): find last matching key in LDML key bag when building KVK

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes: #12056

Will cherry-pick once approved by @srl295 

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts [hunk 1]
```diff
@@ -3,6 +3,23 @@ import { LDMLKeyboard, CompilerCallbacks } from "@keymanapp/developer-utils";
...
 
+// This is a partial polyfill for findLast, so not polluting Array.prototype
+//
+// TODO: remove and replace with Array.prototype.findLast when it is
+// well-supported
+function findLast(arr: any, callback: any) {
+  if (!arr) {
+    return undefined;
+  }
+  const len = arr.length >>> 0;
```
Comment (srl295) [inline developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts:14]: Say what? Is this to normalize the value?

**Review by srl295 — APPROVED** [review]
Seems good. Could also just have a reverse array going in and call find on that.

File: developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts [hunk 2]
```diff
@@ -3,6 +3,23 @@ import { LDMLKeyboard, CompilerCallbacks } from "@keymanapp/developer-utils";
 import { KeysCompiler } from "./keys.js";
 import { LdmlCompilerMessages } from "./ldml-compiler-messages.js";
 
+// This is a partial polyfill for findLast, so not polluting Array.prototype
+//
```
Comment (mcdurdin) [inline developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts:7]:
```suggestion
// https://medium.com/@stheodorejohn/findlast-method-polyfill-in-javascript-bridging-browser-gaps-c3baf6aabae1
```

File: developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts:14]: good question. this is copypasta, so I didn't examine very closely; I found this: https://stackoverflow.com/questions/44946957/what-use-cases-exist-for-using-operator-with-0-unsigned-right-bit-shift

**Review by mcdurdin — COMMENTED** [review]
> Could also just have a reverse array going in and call find on that.

I contemplated doing that but was thinking that this was more forward-looking. Also didn't want to audit for other potential trouble using reversed array :grin:

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.98-alpha](https://keyman.com/downloads/releases/alpha/18.0.98)

**Pull request merged**